src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -1419,7 +1420,12 @@ and tc_abs_expected_function_typ env (bs:binders) t0 (body:term)
           (envbody, letrec_binders, Env.close_guard envbody bs g)
         in
 
-        let envbody, bs, g_env, c, body = check_actuals_against_formals env bs bs_expected body in
+        (* Set letrecs to [] before calling check_actuals_against_formals,
```

Comment:
How come it is necessary now to set letrecs to []. I can well imagine that things get messed up if it is set ... but how was it not causing a problem until now?

Reply:
This might actually be unnecessary now.. I made the change when I tried to check that the `previous_dec` clause was computed at the right arity, throwing an exception for any discrepancy. After doing that, I got the exception in an example similar to this:

```fstar
val list_ref: unit -> u:unit{l_Forall (fun (x:int) -> True)} -> Tot unit
let rec list_ref _ _ = ()
```
The arity is 2, but we descend into the `fun (x:int) -> True` during typechecking and compute a `previous_dec` for it. (The clause cannot be used, so it's a bit of waste, but not immediately a problem.)

I removed the check since it didn't account for mutual recursion (and couldn't easily do so either), but left this change since I think it's anyway right.